server/resp/src/test/java/org/infinispan/server/resp/RespBlockingCommandsTest.java

```diff
@@ -0,0 +1,146 @@
+package org.infinispan.server.resp;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.infinispan.server.resp.test.RespTestingUtil.createClient;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.server.resp.test.RespTestingUtil;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import io.lettuce.core.KeyValue;
+import io.lettuce.core.RedisCommandExecutionException;
+import io.lettuce.core.api.async.RedisAsyncCommands;
+import io.lettuce.core.api.sync.RedisCommands;
+
+/**
+ * Class for blocking commands tests.
+ *
+ * @author Vittorio Rigamonti
+ * @since 15.0
+ */
+@Test(groups = "functional", testName = "server.resp.RespBlockingCommandsTest")
+public class RespBlockingCommandsTest extends SingleNodeRespBaseTest {
+
+   private CacheMode cacheMode = CacheMode.LOCAL;
+   private boolean simpleCache;
+
+   @Factory
+   public Object[] factory() {
+      return new Object[] {
+            new RespBlockingCommandsTest(),
+            new RespBlockingCommandsTest().simpleCache()
+      };
+   }
+
+   RespBlockingCommandsTest simpleCache() {
+      this.cacheMode = CacheMode.LOCAL;
+      this.simpleCache = true;
+      return this;
+   }
+
+   @Override
+   protected String parameters() {
+      return "[simpleCache=" + simpleCache + ", cacheMode=" + cacheMode + "]";
+   }
+
+   @Override
+   protected void amendConfiguration(ConfigurationBuilder configurationBuilder) {
+      if (simpleCache) {
+         configurationBuilder.clustering().cacheMode(CacheMode.LOCAL).simpleCache(true);
+      } else {
+         configurationBuilder.clustering().cacheMode(cacheMode);
+      }
+   }
+
+   @Test
+   void testBlpopAsync() throws InterruptedException, ExecutionException, TimeoutException {
+      RedisAsyncCommands<String, String> redis = redisConnection.async();
+      var client = createClient(30000, server.getPort());
+      RedisAsyncCommands<String, String> redis1 = client.connect().async();
+      redis1.lpush("keyY", "firstY").get();
```

Comment:
I am afraid these async ones with `.get()` would just block until the test suite timeout in case of failures. Looking closely, I think only `redis` needs to be async, and `redis1` could be a `RedisCommands` instead. And we can add some timeout to the `.get()` calls.

Reply:
But the comment I added to the test below should also cover this one.